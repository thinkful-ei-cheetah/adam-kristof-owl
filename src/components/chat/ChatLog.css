.ChatLog{
  width:350px;
  border: 1px solid grey;
  height: 600px;
  overflow-y: auto;
}

