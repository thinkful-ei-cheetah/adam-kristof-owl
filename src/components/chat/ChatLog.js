import React from 'react';
import './ChatLog.css';
export default function ChatLog (props){




  const getName =(id)=>{
 
    const person = props.addInfo.find(person=> person.id === id) 
    return (
    <div>
      <img src={person.avatar} alt = {`img of ${person.name}`}/>
      <span>{person.name}</span>
    </div>
    )

  }

  let getTime= (timeString) =>{
    let time = new Date(timeString);
    return `${time.getHours()}: ${time.getMinutes()}`;
  } 

  const entryLog = props.log.map(logEntry => {
    return (
    <li key={logEntry.timestamp}>
      <div>
        <div>{getName(logEntry.participantId)}</div>

        <div>{getTime(logEntry.timestamp)}</div>
      </div>
    </li>
    )
  })

  return (
    <div className ="ChatLog">
      <ul>
        {entryLog}
      </ul>
    </div>
  )


}