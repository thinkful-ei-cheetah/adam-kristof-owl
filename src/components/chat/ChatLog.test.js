import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './ChatLog'
import renderer from 'react-test-renderer'  


describe('Chat Componenet working', ()=>{
  it('Renders to the page without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<ChatLog log={[]} addInfo={[]}/>, div)
    ReactDOM.unmountComponentAtNode(div)
  });
  
  it('Renders as it did the first time!', ()=>{
    const tree = renderer
    .create(<ChatLog log={[]} addInfo={[]}/>)
    .toJSON()
    expect(tree).toMatchSnapshot()  
  ;
  });
});
