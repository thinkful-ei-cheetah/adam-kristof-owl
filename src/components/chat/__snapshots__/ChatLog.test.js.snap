// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Componenet working Renders as it did the first time! 1`] = `
<div
  className="ChatLog"
>
  <ul />
</div>
`;
